### ISSUE TITLE
Confused about how to optimize

### ISSUE THREAD

**Issue by UpCoder** [yz93/anchor-diff-VOS#3]
Hi, I am confused about how to optimize.
1, what's the loss function? in this paper, it seems not to be mentioned. is it just BCE?
2, what's the ground truth?
Thank you!

**Comment by yz93** [comment]
Dear @UpCoder,

No problem. Thank you for your interest.

1. The loss is indeed binary cross entropy. Use BCEWithLogitsLoss which combines a Sigmoid layer with the BCELoss.

2. The ground truth is 0/1 binary labels, with 1 indicating target and 0 indicating background.

**Comment by UpCoder** [comment]
Thank you for your reply.
So the unsupervised VOS just indicates that you do not use any annotation in the test stage. the semi-supervised method needs the annotation of the first frame in the test stage.
The training processing is fully supervised.
Is it right?

**Comment by yz93** [comment]
Dear @UpCoder,

What you said is exactly right. The use of the word “unsupervised” causes a fair amount of confusion and also some well founded doubts.

By today’s generally accepted definition of “unsupervised”—not using GT labels during training—this may be considered a misuse of a word. But the usage of the word in the sense that you just described I believe is originated from the once canonical setting in VOS, which uses human input as supervision to guide the algorithm at test time, as at those times “supervision” at test time was fairly common.

**Comment by UpCoder** [comment]
OK, got it.


Thank you!

**Comment by mingminzhen** [comment]
@yz93 Hi, for the training step, do you compute the loss for the anchor image, which is not mentioned in the paper?

**Comment by yz93** [comment]
@mingminzhen No. The loss is binary cross-entropy with logits on the output of the network with GT binary labels.

**Comment by mingminzhen** [comment]
@yz93 Hi, for the training step, what is the scale range for randomly resize? If possible, can you provide the data augmentation code?
